sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateSampleDao.java

```diff
@@ -569,7 +569,21 @@ public void setSessionFactory(SessionFactory sessionFactory) {
     criteria.add(Restrictions.or(Restrictions.ilike("externalName", str), Restrictions.ilike("alias", str)));
     @SuppressWarnings("unchecked")
     Collection<Identity> records = criteria.list();
-    // return without fetching all associated SQL items
+    // return without running fetchSqlStore() to get non-Hibernate associated entities
+    return records;
+  }
+
+  @Override
+  public Collection<Identity> getIdentitiesByExternalNameAndProject(String externalName, Long projectId) throws IOException {
+    if (isStringEmptyOrNull(externalName)) return Collections.emptySet();
+    if (projectId == null) throw new IllegalArgumentException("Must provide a projectId in search");
+    String str = DbUtils.convertStringToSearchQuery(externalName);
+    Criteria criteria = currentSession().createCriteria(IdentityImpl.class);
+    criteria.add(Restrictions.eq("project.id", projectId));
+    criteria.add(Restrictions.or(Restrictions.ilike("externalName", str), Restrictions.ilike("alias", str)));
+    @SuppressWarnings("unchecked")
+    Collection<Identity> records = criteria.list();
+    // return without running fetchSqlStore() to get non-Hibernate associated entities
```

Comment:
There are 514 projects with duplicate external names (the BART project is a particularly frequent offender) so there will be times when it returns multiple items.

I will change the fetchSqlStore.